services/replicator/outconn.go

```diff
@@ -153,94 +153,89 @@ func (conn *outConnection) readMsgStream() {
 	var numMsgsRead int32
 
 	for {
-		select {
-		case <-conn.closeChannel:
+		if numMsgsRead >= creditBatchSize {
```

Comment:
I think it will be "cleaner" to move this block of code .. to after the switch block (below) that updates the 'numMsgsRead' counter.